### PR TITLE
openconnect: T4955: Removed wrong authserver in radiusclient.conf

### PR DESCRIPTION
**Pull request by aapostoliuk**
## Change Summary

After merging config dictionary with default values, radius port the default value was merged not in a proper way. It is added as a server.
After creating radiusclient.conf added
and the illegal authserver equal 'port'.
Backported from 1.4

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T4955

## Component(s) name

openconnect

## Proposed changes

After merging config dictionary with default values, radius port the default value was merged not in a proper way.
It is added as a server.
After creating radiusclient.conf added and the illegal authserver equal 'port'.
Deleted the wrong key 'port' in the merged config dictionary
This issue is more complex and described in https://phabricator.vyos.net/T2665
This fix is only a special case of this problem.

## How to test

Configuration:
```
set vpn openconnect authentication mode radius
set vpn openconnect authentication radius server 192.168.192.173 key 'test'
```
Before this fix, radiusclient.conf will be
```
nas-identifier VyOS
authserver 192.168.192.173:1812
authserver port
radius_timeout 2
bindaddr *
servers /run/ocserv/radius_servers
dictionary /etc/radcli/dictionary
default_realm
radius_retries 3
```
After this fix radiusclient.conf will be
```
nas-identifier VyOS
authserver 192.168.192.173:1812
radius_timeout 2
bindaddr *
servers /run/ocserv/radius_servers
dictionary /etc/radcli/dictionary
default_realm
radius_retries 3
```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: src/conf_mode/vpn_openconnect.py [hunk 1]
```diff
@@ -39,6 +39,34 @@
 def get_hash(password):
     return crypt(password, mksalt(METHOD_SHA512))
 
+
+def T2665_default_dict_cleanup(origin: dict, default_values: dict) -> dict:
```
Comment (dmbaturin) [inline src/conf_mode/vpn_openconnect.py]: Should we really reference task numbers in function names?

File: src/conf_mode/vpn_openconnect.py [hunk 2]
```diff
@@ -39,6 +39,34 @@
...
+    :return: merged dict
+    :rtype: dict
+    """
+
+    if 'mode' in origin["authentication"] and "radius" in \
+            origin["authentication"]["mode"]:
+        del origin['authentication']['radius']['server']['port']
+        if not origin["authentication"]['radius']['server']:
+            raise ConfigError(
+                'Openconnect mode radius required at least one radius server')
```
Comment (dmbaturin) [inline src/conf_mode/vpn_openconnect.py]: I think it should be either `OpenConnect authentication mode is set to RADIUS but RADIUS servers are not configured` or `"vpn openconnect authentication mode radius" requires at least one RADIUS server`. Clarity of config errors is important, and so if their formatting.

**Review by dmbaturin — CHANGES_REQUESTED** [review]

File: src/conf_mode/vpn_openconnect.py (hunk 1 above)
Comment (aapostoliuk) [reply src/conf_mode/vpn_openconnect.py]: Changed to _default_dict_cleanup

File: src/conf_mode/vpn_openconnect.py (hunk 2 above)
Comment (aapostoliuk) [reply src/conf_mode/vpn_openconnect.py]: Changed to 'openconnect authentication mode radius requires at least one RADIUS server'

**Review by dmbaturin — APPROVED** [review]

File: src/conf_mode/vpn_openconnect.py [hunk 3]
```diff
@@ -39,6 +39,34 @@
 def get_hash(password):
     return crypt(password, mksalt(METHOD_SHA512))
 
+
+def _default_dict_cleanup(origin: dict, default_values: dict) -> dict:
+    """
+    https://phabricator.vyos.net/T2665
```
Comment (sever-sever) [inline src/conf_mode/vpn_openconnect.py]: Change reference to vyos .dev

**Review by sever-sever — APPROVED** [review]

File: src/conf_mode/vpn_openconnect.py (hunk 3 above)
Comment (aapostoliuk) [reply src/conf_mode/vpn_openconnect.py]: done

**Pull request merged**